document-portal/document-portal-fuse.c

```diff
@@ -3158,10 +3158,16 @@ xdp_fuse_setlk (fuse_req_t req,
                 int sleep)
 {
   const char *op = "SETLK";
+  XdpFile *file = (XdpFile *)fi->fh;
+  int res;
 
   g_debug ("SETLK %lx", ino);
 
-  xdp_reply_err (op, req, ENOSYS);
+  res = fcntl (file->fd, F_SETLK, lock);
+  if (res == 0)
+    return xdp_reply_err (op, req, 0);
+  else
+    return xdp_reply_err (op, req, errno);
```

Comment:
Seems fine to me. I have looked at other functions and the style seems all over the place. So I'll suggest the following:

```c
  res = fcntl (file->fd, F_SETLK, lock);
  if (res != 0)
    return xdp_reply_err (op, req, errno);

  return xdp_reply_err (op, req, 0);
```